service/history/workflowExecutionContext.go

```diff
@@ -120,26 +126,32 @@ func (c *workflowExecutionContext) replicateWorkflowExecution(request *h.Replica
 	lastEventID, transactionID int64) error {
 
 	nextEventID := lastEventID + 1
-	c.msBuilder.ApplyReplicationStateUpdates(request.GetVersion(), lastEventID)
+	c.msBuilder.updateReplicationStateLastEventID(lastEventID)
 	c.msBuilder.executionInfo.NextEventID = nextEventID
 
 	builder := newHistoryBuilderFromEvents(request.History.Events, c.logger)
 	return c.updateHelper(builder, nil, nil, false, true, transactionID)
 }
 
-func (c *workflowExecutionContext) updateWorkflowExecution(transferTasks []persistence.Task,
-	timerTasks []persistence.Task, transactionID int64) error {
-
-	crossDCEnabled := c.shard.GetService().GetClusterMetadata().IsGlobalDomainEnabled()
-	if crossDCEnabled {
+func (c *workflowExecutionContext) updateVersion() error {
+	if c.msBuilder.replicationState != nil {
```

Comment:
Same thing here.  Just to be sure wrap this check with  'IsGlobalDomainEnabled'.